File: app/src/main/java/de/exercicse/jrossbach/podcaster/feature/search/ui/PodcastChannelView.java
package de.exercicse.jrossbach.podcaster.feature.search.ui;


import java.util.List;

public interface PodcastChannelView {

    void onItemsLoadedSuccessfully(List<PodcastItemViewModel> podcastItemViewModels);
    void showProgress(boolean show);
    void onItemClick(PodcastItemViewModel podcastItemViewModel);
    void showError(String message);
}

File: app/src/main/java/de/exercicse/jrossbach/podcaster/feature/channel/ui/PodcastChannelView.kt
package de.exercicse.jrossbach.podcaster.feature.channel.ui

import de.exercicse.jrossbach.podcaster.base.ui.BaseView
import de.exercicse.jrossbach.podcaster.feature.channel.data.PodcastResponse

interface PodcastChannelView: BaseView {
    fun onChannelLoaded(podcast: PodcastResponse)
}